tsl/org.testeditor.tsl.dsl.tests/src/org/testeditor/tsl/dsl/tests/AbstractTest.xtend

```diff
@@ -40,6 +42,12 @@ abstract class AbstractTest extends AbstractXtextTests {
 		val injector = createInjector
 		setInjector(injector)
 		injector.injectMembers(this)
+		
+		if( !EPackage.Registry.INSTANCE.containsKey("http://www.testeditor.org/tsl")){
```

Comment:
Those lines should not be in the test class, they should go into `TslStandaloneSetup`. Overwrite the `createInjectorAndDoEMFRegistration` and put it there before calling the super method. Additionally, the namespace uri can be referenced better:

	override Injector createInjectorAndDoEMFRegistration() {
		if (!EPackage.Registry.INSTANCE.containsKey(TslPackage.eNS_URI)) {
			EPackage.Registry.INSTANCE.put(TslPackage.eNS_URI, TslPackage.eINSTANCE)
		}
		return super.createInjectorAndDoEMFRegistration
	}